heltour/tournament/views.py

```diff
@@ -1234,56 +1234,42 @@ def _view(league_tag, season_tag, user_data):
 
 
 class ActivePlayerTableView(LeagueView):
-    @cached_as(League, Season, Round)
+    @cached_as(PlayerPairing)
     def view(self, page: int = 1):
-        tablesums = Player.objects.only("lichess_username").annotate(
-            blackcount=Count("pairings_as_black", filter=self.black_games_Q(), distinct=True),
-            whitecount=Count("pairings_as_white", filter=self.white_games_Q(), distinct=True),
-            game_count=F("blackcount") + F("whitecount"),
-            season_count=Count("seasonplayer", filter=self.seasons_Q(), distinct=True),
-            latest_season=Subquery(self.latest_season_subquery()),
-        ).order_by("-game_count", "season_count")
+        tablesums = self.league.get_active_players()
 
         paginator = Paginator(tablesums, DEFAULT_PAGE_SIZE)
         page_obj = paginator.get_page(page)
 
+        pks = [player.player_id for player in page_obj.object_list]
+        seasondatafull = SeasonPlayer.objects.filter(player__pk__in = pks, season__league=self.league).values("player__pk").annotate(
+                season_count = Count("player__pk"),
+                ).values("player__lichess_username", "player__pk", "season_count")
+
+        seasondata = {f'{pl["player__pk"]}': [pl["player__lichess_username"], pl["season_count"]] for pl in seasondatafull}
```

Comment:
why convert to a string here?

Reply:
the key of the dict can be an int

Reply:
good questions. i think i initially had the username as the key, and then instead of removing the string conversion here, added another conversion below.